core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -346,19 +349,15 @@ private InputStream getUncompressedBody() {
   }
 
   private InputStream getCompressedBody() {
-    if (compression == Compression.NONE) {
+    if (decompressor == MessageEncoding.NONE) {
       throw Status.INTERNAL.withDescription(
           "Can't decode compressed frame as compression not configured.").asRuntimeException();
     }
 
-    if (compression != Compression.GZIP) {
-      throw new AssertionError("Unknown compression type");
-    }
-
     try {
-      return new GZIPInputStream(ReadableBuffers.openStream(nextFrame, true));
+      return decompressor.decompress(ReadableBuffers.openStream(nextFrame, true));
     } catch (IOException e) {
-      throw new RuntimeException(e);
+      throw Throwables.propagate(e);
```

Comment:
Yeah, IIRC there is an effort across g3 to remove references to `Throwables.propagate` as well.